packages/webpack-cli/lib/webpack-cli.js

```diff
@@ -245,53 +248,73 @@ class WebpackCLI extends GroupHelper {
     async run(args, cliOptions) {
         await this.processArgs(args, cliOptions);
 
-        const compiler = this.createCompiler(this.compilerConfiguration);
-
-        const options = this.compilerConfiguration;
-        const outputOptions = this.outputConfiguration;
+        let options = this.compilerConfiguration;
+        let outputOptions = this.outputConfiguration;
 
-        if (outputOptions.interactive) {
-            const interactive = require('./utils/interactive');
+        let isWatchMode = false;
 
-            return interactive(compiler, options, outputOptions);
+        if (Array.isArray(options)) {
+            isWatchMode = options.some((options) => options.watch);
+        } else {
+            isWatchMode = options.watch;
         }
 
-        const compilers = compiler.compilers ? compiler.compilers : [compiler];
-        const isWatchMode = Boolean(compilers.find((compiler) => compiler.options.watch));
-        const isRawOutput = typeof outputOptions.json === 'undefined';
+        const PluginName = 'webpack-cli';
 
-        if (isRawOutput) {
-            for (const compiler of compilers) {
-                if (outputOptions.progress) {
-                    const { ProgressPlugin } = webpack;
+        class WebpackCLIPlugin {
```

Comment:
Can we move this plugin to `plugins` (on the same level with utils directory) directory for better reading?

Reply:
Yes 👍